Auto-Deployment.md

```diff
@@ -0,0 +1,72 @@
+      ```
+      gcloud kms keyrings add-iam-policy-binding amp-github-apps-keyring \
+        --location=global \
+        --member group:amp-core-infra@google.com \
+        --role roles/cloudkms.cryptoKeyDecrypter
+      ```
+      > This will allow all members of the group to encrypt and decrypt secrets using this keyring.
+
+5. Create the Cloud KMS CryptoKey
+      ```
+      gcloud kms keys create app-env-key \
+        --location=global \
+        --keyring=amp-github-apps-keyring \
+        --purpose=encryption
+      ```
+
+6. Grant decryption permissions to the project service account
+    ```
+    gcloud kms keys add-iam-policy-binding app-env-key \
+      --location=global \
+      --keyring=amp-github-apps-keyring \
+      --member=serviceAccount:[PROJECT_NUMBER]@cloudbuild.gserviceaccount.com \
+      --role=roles/cloudkms.cryptoKeyDecrypter
+    ```
+
+7. Create the Cloud Build configuration file `cloud_build.yaml`
+    - Use an existing configuration as a template
+    - The first two steps must be installing `yarn` and replacing secrets
+    - The remaining steps will parallel the manual deployment process, such as `npm install`, `gcloud app deploy`, etc.
+
+8. Grant App Engine access to Cloud Build service account on the [Service account permissions](https://console.cloud.google.com/cloud-build/settings) page
+    - Set _App Engine Admin_ to _Enable_
+
+9. Create the [Cloud Build Trigger](https://pantheon.corp.google.com/cloud-build/triggers)
+    - Click _Connect Repository_ and follow the steps to connect to `ampproject/amp-github-apps`
+    - Create a trigger named "on-deploy-tag", select `ampproject/amp-github-apps` as the _Source_, and set the _Event_ to _Push new tag_
+    - For the tag pattern, use `deploy-{your-app-name}-\d{14}` (ex. `deploy-bundle-size-20200122154000` would be the deploy tag for the `bundle-size` app, with a timestamp of 15:40:00 on 2020-01-22)
+    - Under _Build Configuration_, select _Cloud Build configuration file_ and provide the path to your Cloud Build file, ex. `bundle-size/cloud_build.yaml`
+10. Add the NPM script `deploy-tag`, which creates a git tag in the proper tag format for your app
+    - Ex. ``git tag 'deploy-bundle-size-'`date '+%Y%m%d%H%M%S'` ``
+
+## To store an encrypted secret
+
+1. Add the environment variable to `redacted.env`
+    > Note: We commit `redacted.env` with all non-sensitive environment variables and create `.env` on the GCloud Build instance at deployment. This is to prevent accidentally committing a secret token.
+2. Encrypt and base64-encode the secret by running `build-system/encrypt-secrets.sh`
+3. Add the environment variable name and the base64-encoded value from above to the `secrets.secretEnv` dict in `cloud_build.yaml`
+4. Add the environment variable name to the `secretEnv` field of the `replace-secrets` step in `cloud_build.yaml`
+    > This provides the decrypted secret to the environment so `replace-scripts` can construct the un-redacted `.env` file.
```

Comment:
Do you mean encrypted secret, not decrypted?

Reply:
Noop; on the build system, Cloud Build provides the decrypted secret to the environment.

Reply:
Added some clarifying language.

Reply:
How about this if it's accurate:
"This allows Cloud Build to decrypt the env vars and construct the un-redacted .env file"

Reply:
Not sure if this comment came before or after you saw the most recent update, PTAL